scionlab/models/core.py

```diff
@@ -312,23 +274,22 @@ def isd_as_path_str(self):
 
     def init_keys(self):
         """
-        Initialise signing and encryption key pairs, the MasterASKey (used for
-        hop field Message Authentication Codes (MAC) and the Dynamically
-        Recreatable Keys (DRKeys)).
-        If this is a core AS, also initialise the core AS signing key pairs.
+        Initialise AS signing and encryption keys.
+        If this is a core AS, also initialise the core AS voting and issuing keys.
+
+        Note: does not set master_as_key.
```

Comment:
Or is the reason for this that we can reset the pki in 'migrations/0006_pki.py', while keeping the same master AS key?

Reply:
The `master_as_key` is a string field on the AS object. To set this, we need to modify and save the AS model. The most straight-forward way to initialize this is to just set it directly in the `create` call.
All the other keys are separate objects with an FK to AS. Creating these does not require changing the AS object, but, it can _only_ be done after the AS object is created (need to have a DB primary key).

But to initialize both master key and the Key objects in the same function, we'd first have to create the AS model (to be able to create the Keys) and then immediately update it again to set the master key. This would seem to be _less_ clear.

The master AS key is kind of a second-class citizen (both in our and in the SCION codebase), so to me this seemed ok-ish. However, if you feel that this is very confusing, I'll change it. 

No, the migration thing is unrelated, we could safely recreate the master key during the migration.